internal/rsl/rsl.go

```diff
@@ -411,6 +423,30 @@ func GetLatestReferenceEntryForRefBefore(repo *git.Repository, refName string, a
 	return targetEntry, annotations, nil
 }
 
+// GetLatestUnskippedReferenceEntryForRef returns the latest reference entry for
+// the ref that does not have an annotation marking it as to-be-skipped.
+func GetLatestUnskippedReferenceEntryForRef(repo *git.Repository, refName string) (*ReferenceEntry, []*AnnotationEntry, error) {
+	return GetLatestUnskippedReferenceEntryForRefBefore(repo, refName, plumbing.ZeroHash)
+}
+
+// GetLatestUnskippedReferenceEntryForRefBefore returns the first reference
+// entry for the ref before the anchor that does not have an annotation marking
+// it as to-be-skipped.
+func GetLatestUnskippedReferenceEntryForRefBefore(repo *git.Repository, refName string, anchor plumbing.Hash) (*ReferenceEntry, []*AnnotationEntry, error) {
+	for {
+		latestEntry, annotations, err := GetLatestReferenceEntryForRefBefore(repo, refName, anchor)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if !latestEntry.SkippedBy(annotations) {
+			return latestEntry, annotations, nil
+		}
+
+		anchor = latestEntry.ID
```

Comment:
I think we should only do this if latest entry != anchor? IIUC if the anchor itself is skipped we'll iterate past it until we find the first non-skipped id.

Reply:
latestEntry can't be anchor because we use `GetLatestReferenceEntryForRefBefore(repo, refName, anchor)` to get it. So we know that's always true, right?